src/client/dbif.cc

```diff
@@ -0,0 +1,943 @@
+/*
+ * Copyright 2017 CodiLime
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <memory>
+#include <vector>
+
+#include <QSharedPointer>
+
+#include "data/types.h"
+#include "network/msgpackwrapper.h"
+#include "client/dbif.h"
+
+namespace veles {
+namespace client {
+
+/*****************************************************************************/
+/* NCObjectHandle */
+/*****************************************************************************/
+
+NCObjectHandle::NCObjectHandle(NCWrapper* nc, data::NodeID id,
```

Comment:
since this takes ownership of nc, I think parent of it should be changed to this